services/QuillLMS/app/services/dual_google_id_and_clever_id_resolver.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class DualGoogleIdAndCleverIdResolver < ApplicationService
+  attr_reader :user
+
+  RESOLVERS = %i[
+    resolve_by_account_type
+    resolve_by_auth_credential
+    resolve_by_last_classroom
+  ].freeze
+
+
+  def initialize(user)
+    @user = user
+  end
+
+  def run
+    RESOLVERS.detect { |resolver| send(resolver) }
+  end
+
+  private def log_account_type_change(changed_attribute, previous_value, action)
+    ChangeLog.create(
+      action: action,
+      changed_attribute: changed_attribute,
+      changed_record: user,
+      explanation: caller_locations[0].to_s,
+      previous_value: previous_value
+    )
+  end
+  private def resolve_by_account_type
```

Comment:
Not that it matters, but missing newline space separating this method from the previous one?

Reply:
That's really ugly.  Thanks for flagging this.